packages/astro/src/@types/astro.ts

```diff
@@ -371,19 +371,20 @@ type ServerConfig = {
 
 	/**
 	 * @name server.open
-	 * @type {boolean}
+	 * @type {string | boolean}
 	 * @default `false`
 	 * @version 2.1.8
 	 * @description
 	 * Control whether the dev server should open in your browser window on startup.
+	 * A URL string can also be passed to specify the URL to open.
 	 *
 	 * ```js
 	 * {
 	 *   server: { open: true }
```

Comment:
I believe this documentation should be updated and provide an example of how to use the new variant string. Plus, we should explain how this works. Can a user provide any kind of string? Should a user expect an error if a string doesn't have a certain format?

I noticed that you added a new paragraph, but it goes in conflict with your testing block, where you used `/test`, which is not a full URL

Reply:
You can pass either a pathname (so the origin is based on `host`), or the full URL path here. I'll update this part to clarify it 👍

Reply:
I added some examples, but I didn't explain what would happen if you e.g. pass an invalid URL. Since the `site` option also doesn't explain it, I think it should implicitly mean that it'll fail with `new URL` if you pass an invalid URL.